corehq/apps/reports/tests/test_util.py

```diff
@@ -1,24 +1,49 @@
-import os
 from unittest.mock import patch, Mock
+from datetime import datetime
+import pytz
 
 from django.core.cache import cache
 from django.test import SimpleTestCase
 
 from testil import eq
+from freezegun import freeze_time
 
 from corehq.apps.reports.tasks import summarize_user_counts
-from corehq.apps.reports.util import get_user_id_from_form
+from corehq.apps.reports.util import get_user_id_from_form, datespan_from_beginning
 from corehq.form_processor.exceptions import XFormNotFound
 from corehq.form_processor.models import XFormInstance
 from corehq.form_processor.utils import TestFormMetadata
-from corehq.util.test_utils import TestFileMixin, get_form_ready_to_save
+from corehq.util.test_utils import get_form_ready_to_save
 from corehq.apps.reports.standard.cases.utils import get_user_type
 from corehq.apps.data_interfaces.deduplication import DEDUPE_XMLNS
+from corehq.apps.domain.models import Domain
 
 DOMAIN = 'test_domain'
 USER_ID = "5bc1315c-da6f-466d-a7c4-4580bc84a7b9"
 
 
+@freeze_time("2023-02-10")
+class TestDateSpanFromBeginning(SimpleTestCase):
+    def test_start_and_end_date_dont_change_in_utc(self):
+        start_time = datetime(year=2020, month=4, day=5)
+
+        domain = Domain(date_created=start_time)
+        datespan = datespan_from_beginning(domain, pytz.utc)
+
+        self.assertEqual(datespan.startdate, datetime(year=2020, month=4, day=5))
+        self.assertEqual(datespan.enddate, datetime(year=2023, month=2, day=10))
+
+    def test_start_and_end_date_can_change_if_behind_utc(self):
+        start_time = datetime(year=2020, month=4, day=5)
+
+        domain = Domain(date_created=start_time)
+        # for some reason, pytz represents GMT+1 as 1 hour BEHIND GMT.
```

Comment:
This isn't pytz, it is because the `Etc` zones use POSIX conventions:

> Be consistent with POSIX TZ settings in the Zone names,
> even though this is the opposite of what many people expect.
> POSIX has positive signs west of Greenwich, but many people expect
> positive signs east of Greenwich.  For example, TZ='Etc/GMT+4' uses
> the abbreviation "-04" and corresponds to 4 hours behind UT
> (i.e. west of Greenwich) even though many people would expect it to
> mean 4 hours ahead of UT (i.e. east of Greenwich).

https://github.com/eggert/tz/blob/main/etcetera#L37-L43

Reply:
Thanks @snopoke ! Useful context. There isn't an alternative I can use that is more aligned with what people expect, right?

Reply:
Initially I was going to use a real timezone, but didn't feel it was as intuitive to explain my intent as UTC-1.  The custom timezone thing is closer to what I was looking for, but I think the verbosity detracts from my goal of making it quickly understood, so I think I'm reluctantly leaving it as GMT+1 for now.